src/test/java/io/strimzi/kafka/bridge/mqtt/mapper/MqttKafkaRegexMapperTest.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.bridge.mqtt.mapper;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.PatternSyntaxException;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertThrows;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.hamcrest.CoreMatchers.startsWith;
+
+/**
+ * Unit tests for {@link MqttKafkaRegexMapper}
+ */
+public class MqttKafkaRegexMapperTest {
+
+
+    /**
+     * Test for default topic.
+     * If the MQTT topic does not match any of the mapping rules, the default topic is used.
+     * E.g. if the Topic Mapping Rules is empty, the default topic is used by default.
+     */
+    @Test
+    public void testDefaultTopic() {
+        List<MappingRule> rules = new ArrayList<>();
+
+        MqttKafkaRegexMapper mapper = new MqttKafkaRegexMapper(rules);
+
+        MappingResult result = mapper.map("sensor/temperature");
+        assertThat("Should use the default topic when no mapping pattern matches.",
+                result.kafkaTopic(), is(MqttKafkaSimpleMapper.DEFAULT_KAFKA_TOPIC));
+
+        assertThat("The key for the default topic should be null",
+                result.kafkaKey(), nullValue());
+    }
+
+    /**
+     * Test the mapping of single level topics.
+     */
+    @Test
+    public void testSingleLevel() {
+        List<MappingRule> rules = new ArrayList<>();
+
+        rules.add(new MappingRule("sensors/[^/]+/data", "sensor_data", null));
+        rules.add(new MappingRule("devices/([^/]+)/data", "devices_$1_data", null));
+        rules.add(new MappingRule("fleet/([0-9]+)/vehicle/(\\w+)", "fleet_$1", "vehicle$2"));
+        rules.add(new MappingRule("building/(\\d+)/floor/(\\d+).*", "building_$1", "floor_$2"));
+        rules.add(new MappingRule("term/(\\d+)", "term$1", null));
+        rules.add(new MappingRule("devices/([^/]+)/data/(\\b(all|new|old)\\b)", "devices_$1_data", "devices_$2"));
+
+        MqttKafkaRegexMapper mapper = new MqttKafkaRegexMapper(rules);
+
+        // Test building/(\\d+)/floor/(\\d+).*
+        MappingResult mappingResult = mapper.map("building/14/floor/25");
+
+        assertThat("building/(\\d+)/floor/(\\d+).* should be mapped to building_$1",
+                mappingResult.kafkaTopic(), is("building_14"));
+
+        assertThat("The key for building_$1 should be expanded to room_25",
+                mappingResult.kafkaKey(), is("room_25"));
```

Comment:
```suggestion
                mappingResult.kafkaKey(), is("floor_25"));
```

Reply:
Yeah the test is failing locally. The Azure pipeline will help on next PRs ;-)